File: src/dataset/c/prime_factors_Exercism_chriswilding/openAI/ChatGPT4_prime_factors_v3.c
#include "prime_factors.h"

#include <math.h>

size_t find_factors(uint64_t n, uint64_t factors[static MAXFACTORS]) {
  size_t i = 0;

  while ((n & 1) == 0) {
    factors[i++] = 2;
    n >>= 1;
  }

  for (uint64_t j = 3; j * j <= n; j += 2) {
    while (n % j == 0) {
      factors[i++] = j;
      n /= j;
    }
  }

  if (n > 1) {
    factors[i++] = n;
  }

  return i;
}

File: src/dataset/javascript/prime-factors_exercism-javascript-ThomasZumsteg/openAI/ChatGPT4_prime-factors_v3.js
function primeFactors(remainder) {
    let factor = 2;
    const factors = [];
    while (remainder >= factor * factor) {
        if (remainder % factor === 0) {
            factors.push(factor);
            remainder /= factor;
        } else {
            factor += factor === 2 ? 1 : 2;
        }
    }
    if (remainder > 1) factors.push(remainder);
    return factors;
}

export default { for: primeFactors };